ouroboros-network-api/src/Ouroboros/Network/NodeToNode/Version.hs

```diff
@@ -149,35 +154,79 @@ nodeToNodeCodecCBORTerm version
                              InitiatorOnlyDiffusionMode         -> True
                              InitiatorAndResponderDiffusionMode -> False)
               , CBOR.TInt (case peerSharing of
-                             NoPeerSharing      -> 0
-                             PeerSharingPrivate -> 1
-                             PeerSharingPublic  -> 2)
+                             PeerSharingDisabled -> 0
+                             PeerSharingEnabled  -> 1)
               , CBOR.TBool query
               ]
 
         decodeTerm :: NodeToNodeVersion -> CBOR.Term -> Either Text NodeToNodeVersionData
         decodeTerm _ (CBOR.TList [CBOR.TInt x, CBOR.TBool diffusionMode, CBOR.TInt peerSharing, CBOR.TBool query])
           | x >= 0
           , x <= 0xffffffff
-          = case peerSharing of
-            0 -> good NoPeerSharing
-            1 -> good PeerSharingPrivate
-            2 -> good PeerSharingPublic
-            _ -> bad
-          | otherwise -- x < 0 || x > 0xffffffff
-          = Left $ T.pack $ "networkMagic out of expected range [0..ffffffff]: " <> show x
-          where
-            good sharing
-              = Right
-                  NodeToNodeVersionData {
-                      networkMagic = NetworkMagic (fromIntegral x),
-                      diffusionMode = if diffusionMode
-                                      then InitiatorOnlyDiffusionMode
-                                      else InitiatorAndResponderDiffusionMode,
-                      peerSharing = sharing,
-                      query = query
-                    }
-            bad = Left $ T.pack $ "peerSharing out of expected range [0..2]: " <> show peerSharing
+          , Just ps <- case peerSharing of
+                        0 -> Just PeerSharingDisabled
+                        1 -> Just PeerSharingEnabled
+                        _ -> Nothing
+          = Right
+              NodeToNodeVersionData {
+                  networkMagic = NetworkMagic (fromIntegral x),
+                  diffusionMode = if diffusionMode
+                                  then InitiatorOnlyDiffusionMode
+                                  else InitiatorAndResponderDiffusionMode,
```

Comment:
This is out of the scope, but we should rename `diffusionMode` field as `initiatorOnlyMode` otherwise it's a bit confusing. I created an issue for it #4708.

Reply:
Thank you!